flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/assigner/SplitAssigner.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.source.assigner;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nullable;
+import org.apache.flink.api.connector.source.SplitEnumeratorContext;
+import org.apache.iceberg.flink.source.split.IcebergSourceSplit;
+import org.apache.iceberg.flink.source.split.IcebergSourceSplitState;
+
+/**
+ * SplitAssigner interface is extracted out as a separate component so that
+ * we can plug in different split assignment strategy for different requirements. E.g.
+ * <ul>
+ * <li> Simple assigner with no ordering guarantee or locality aware optimization.</li>
+ * <li> Locality aware assigner that prefer splits that are local.</li>
+ * <li> Snapshot aware assigner that assign splits based on the order they are committed.</li>
+ * <li> Event time alignment assigner that assign splits satisfying certain time ordering
+ *      within a single source or across sources.</li>
+ * </ul>
+ *
+ * <p>
+ * Enumerator should call the assigner APIs from the coordinator thread.
+ * This is to simplify the thread safety for assigner implementation.
+ */
+public interface SplitAssigner extends AutoCloseable {
```

Comment:
Most of the time, we use `Closeable` rather than `AutoCloseable` because many libraries check for `Closeable`. `Closeable` implements `AutoCloseable`, but not the other way around so it is usually safer to use `Closeable` even if it isn't as accurate.